### PR TITLE
update for third party tools supported by TiDB

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What is changed, added or deleted? (Required)

update for third party tools supported by TiDB

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

N/A

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-supported-by-pingcap.md [hunk 1]
```diff
@@ -13,55 +13,62 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
 PingCAP 与开源社区合作，通过三方工具提供以下支持：
 
 - Full: 表明 PingCAP 将尽力支持该工具的绝大多数功能兼容性。将定期地对下表中记录的最新版本进行测试。
-- Beta: 表名 PingCAP 正在努力支持该工具。该工具的主要功能将与 TiDB 兼容(连接数据库及基本数据库操作)，但有可能会出现意外行为，且可能需要额外的步骤进行集成。
+- Beta: 表明 PingCAP 正在努力为该工具提供全面支持。该工具的主要功能可以和 TiDB 兼容（例如连接和数据库的基本操作），但缺乏对该工具所有功能的支持，有可能出现一些意外的行为。
```
Comment (Oreoxmt) [inline develop/dev-guide-supported-by-pingcap.md:16]:
```suggestion
- Beta：表明 PingCAP 正在努力为该工具提供全面支持。该工具的主要功能可以和 TiDB 兼容（例如连接和数据库的基本操作），但缺乏对该工具所有功能的支持，有可能出现一些意外的行为。
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 2]
```diff
@@ -13,55 +13,62 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
+
+> **注意：**
+>
+> 除非明确说明，否则对于支持的 driver 或者 ORM 框架并不包括[应用端事务重试和错误处理](/develop/dev-guide-transaction-troubleshoot.md#应用端重试和错误处理)。
```
Comment (Oreoxmt) [inline develop/dev-guide-supported-by-pingcap.md:20]:
```suggestion
> 除非明确说明，否则对于支持的 Driver 或者 ORM 框架并不包括[应用端事务重试和错误处理](/develop/dev-guide-transaction-troubleshoot.md#应用端重试和错误处理)。
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 3]
```diff
@@ -13,55 +13,62 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
 PingCAP 与开源社区合作，通过三方工具提供以下支持：
 
 - Full: 表明 PingCAP 将尽力支持该工具的绝大多数功能兼容性。将定期地对下表中记录的最新版本进行测试。
```
Comment (Oreoxmt) [inline develop/dev-guide-supported-by-pingcap.md:15]:
```suggestion
- Full：表明 PingCAP 将尽力支持该工具的绝大多数功能兼容性。将定期地对下表中记录的最新版本进行测试。
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 4]
```diff
@@ -12,56 +12,63 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
+| C#(.Net)   | [MySQL Connector/ODBC](https://downloads.mysql.com/archives/c-odbc/)     | 8.0.28  | Beta | N/A                                                                                        | N/A                                                                            |
+| Go         | [go-sql-driver/mysql](https://github.com/go-sql-driver/mysql)            | v1.6.0  | Full | N/A                                                                                        | [TiDB 和 Golang 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-golang.md) |
+| Java       | [JDBC](https://dev.mysql.com/downloads/connector/j/)                     | 5.1.46  | Full | N/A                                                                                        | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md)     |
+| Java       | [JDBC](https://dev.mysql.com/downloads/connector/j/)                     | 8.0.29  | Full | [pingcap/mysql-connector-j](https://github.com/pingcap/mysql-connector-j/tree/release/8.0) | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md)     |
```
Comment (Icemap) [inline develop/dev-guide-supported-by-pingcap.md:33]: https://github.com/pingcap/mysql-connector-j/tree/release/8.0
这个好像没有上传到 Maven 中央库吧，这样 Java 很难使用到的。
我们之前有传过一些库的：https://mvnrepository.com/artifact/com.pingcap.tidb

File: develop/dev-guide-supported-by-pingcap.md [hunk 5]
```diff
@@ -12,56 +12,63 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
 
-|                  GUI                   | 最新已测试版本 | 支持等级 | 教程地址 |
-| :------------------------------------: | :------------: | :------: | :------: |
-| [Navicat](https://www.navicat.com/en/) |
+| GUI                                           | 最新测试版本  | 支持级别 | 教程  |
+|-----------------------------------------------|---------|------|-----|
+| [DBeaver](https://dbeaver.io/)                | 22.1.0  | Beat | N/A |
+| [Navicat for MySQL](https://www.navicat.com/) | 16.0.14 | Beat | N/A |
 
 ## IDE
```
Comment (Icemap) [inline develop/dev-guide-supported-by-pingcap.md:73]:
```suggestion
```

没有的话就先隐藏吧

File: develop/dev-guide-supported-by-pingcap.md [hunk 6]
```diff
@@ -12,56 +12,63 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
+| Go                    | [upper/db](https://github.com/upper/db)                                                                                                                                       | v4.5.2      | Full | N/A                                                    | N/A                                                                            |
+| Go                    | [xorm](https://gitea.com/xorm/xorm)                                                                                                                                           | v1.3.1      | Full | N/A                                                    | N/A                                                                            |
+| Java                  | [Hibernate](https://hibernate.org/orm/) (including [Hibernate Spatial](https://docs.jboss.org/hibernate/orm/current/userguide/html_single/Hibernate_User_Guide.html#spatial)) | 5.6.5.Final | Beta | N/A                                                    | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md)     |
```
Comment (Icemap) [inline develop/dev-guide-supported-by-pingcap.md:46]:
```suggestion
| Java                  | [Hibernate](https://hibernate.org/orm/) | 5.6.5.Final | Beta | N/A                                                    | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md)     |
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 7]
```diff
@@ -12,56 +12,63 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
+| JavaScript/TypeScript | [TypeORM](https://www.npmjs.com/package/typeorm)                                                                                                                              | v0.3.6      | Beta | N/A                                                    | N/A                                                                            |
+| PHP                   | [laravel](https://laravel.com/)                                                                                                                                               | v9.1.10     | Beta | [laravel-tidb](https://github.com/colopl/laravel-tidb) | N/A                                                                            |
+| Python                | [Django](https://pypi.org/project/Django/) (including [GeoDjango](https://docs.djangoproject.com/en/4.0/ref/contrib/gis/)  )                                                  | v4.0.5      | Beta | N/A                                                    | N/A                                                                            |
```
Comment (Icemap) [inline develop/dev-guide-supported-by-pingcap.md:53]:
```suggestion
| Python                | [Django](https://pypi.org/project/Django/)                                                  | v4.0.5      | Beta | N/A                                                    | N/A                                                                            |
```

**Review by Icemap — COMMENTED** [review]
Rest LGTM

File: develop/dev-guide-supported-by-pingcap.md [hunk 8]
```diff
@@ -12,56 +12,63 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
+| Go                    | [xorm](https://gitea.com/xorm/xorm)                                                                                                                                           | v1.3.1      | Full | N/A                                                    | N/A                                                                            |
+| Java                  | [Hibernate](https://hibernate.org/orm/) (including [Hibernate Spatial](https://docs.jboss.org/hibernate/orm/current/userguide/html_single/Hibernate_User_Guide.html#spatial)) | 5.6.5.Final | Beta | N/A                                                    | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md)     |
+| Java                  | [mybatis](https://mybatis.org/mybatis-3/)                                                                                                                                     | v3.5.10     | Beta | N/A                                                    | N/A                                                                            |
```
Comment (Icemap) [inline develop/dev-guide-supported-by-pingcap.md:47]:
```suggestion
| Java                  | [mybatis](https://mybatis.org/mybatis-3/)                                                                                                                                     | v3.5.10     | Full | N/A                                                    | N/A                                                                            |
```

File: develop/dev-guide-supported-by-pingcap.md (hunk 4 above)
Comment (Icemap) [reply develop/dev-guide-supported-by-pingcap.md:33]: 容后处理。

**Review by Icemap — APPROVED** [review]

File: develop/dev-guide-supported-by-pingcap.md [hunk 9]
```diff
@@ -12,58 +12,64 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
 
-|                  GUI                   | 最新已测试版本 | 支持等级 | 教程地址 |
-| :------------------------------------: | :------------: | :------: | :------: |
-| [Navicat](https://www.navicat.com/en/) |
+| GUI                                           | 最新测试版本  | 支持级别 | 教程  |
+|-----------------------------------------------|---------|------|-----|
+| [DBeaver](https://dbeaver.io/)                | 22.1.0  | Beat | N/A |
+| [Navicat for MySQL](https://www.navicat.com/) | 16.0.14 | Beat | N/A |
 
-## IDE
```
Comment (Oreoxmt) [inline develop/dev-guide-supported-by-pingcap.md:73]:
```suggestion
```

**Comment by TomShawn** [comment]
Does release-6.1 need this update?

**Comment by Icemap** [comment]
Yep. It is non-related to the version. And we add this to TiDB integration test. From now on, every TiDB PR will run these ORM / Driver tests. @TomShawn

File: develop/dev-guide-supported-by-pingcap.md [hunk 10]
```diff
@@ -12,58 +12,63 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
+| Go                    | [xorm](https://gitea.com/xorm/xorm)                                                                                                                                           | v1.3.1      | Full | N/A                                                    | N/A                                                                            |
+| Java                  | [Hibernate](https://hibernate.org/orm/) | 6.1.0.Final | Full | N/A                                                    | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md)     |
+| Java                  | [mybatis](https://mybatis.org/mybatis-3/)                                                                                                                                     | v3.5.10     | Full | N/A                                                    | N/A                                                                            |
```
Comment (Icemap) [inline develop/dev-guide-supported-by-pingcap.md:47]:
```suggestion
| Java                  | [mybatis](https://mybatis.org/mybatis-3/)                                                                                                                                     | v3.5.10     | Full | N/A                                                    | N/A                                                                            |
| jOOQ                  | [jOOQ](https://github.com/jOOQ/jOOQ)                                                                                                                                     | v3.16.7 (Open Source)     | Full | N/A                                                    | N/A                                                                            |
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 11]
```diff
@@ -12,58 +12,64 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
 ## GUI
 
-|                  GUI                   | 最新已测试版本 | 支持等级 | 教程地址 |
-| :------------------------------------: | :------------: | :------: | :------: |
-| [Navicat](https://www.navicat.com/en/) |
-
-## IDE
+| GUI                                           | 最新测试版本  | 支持级别 | 教程  |
+|-----------------------------------------------|---------|------|-----|
+| [DBeaver](https://dbeaver.io/)                | 22.1.0  | Beat | N/A |
```
Comment (winkyao) [inline develop/dev-guide-supported-by-pingcap.md:71]:
```suggestion
| [DBeaver](https://dbeaver.io/)                | 22.1.0  | Beta | N/A |
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 12]
```diff
@@ -12,58 +12,64 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
+| [Navicat for MySQL](https://www.navicat.com/) | 16.0.14 | Beat | N/A |
```
Comment (winkyao) [inline develop/dev-guide-supported-by-pingcap.md:72]:
```suggestion
| [Navicat for MySQL](https://www.navicat.com/) | 16.0.14 | Beta | N/A |
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 13]
```diff
@@ -12,58 +12,64 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
+| C#(.Net)   | [MySQL Connector/ODBC](https://downloads.mysql.com/archives/c-odbc/)     | 8.0.28  | Beta | N/A                                                                                        | N/A                                                                            |
+| Go         | [go-sql-driver/mysql](https://github.com/go-sql-driver/mysql)            | v1.6.0  | Full | N/A                                                                                        | [TiDB 和 Golang 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-golang.md) |
+| Java       | [JDBC](https://dev.mysql.com/downloads/connector/j/)                     | 5.1.46  | Full | N/A                                                                                        | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md)     |
+| Java       | [JDBC](https://dev.mysql.com/downloads/connector/j/)                     | 8.0.29  | Full | [pingcap/mysql-connector-j](https://github.com/pingcap/mysql-connector-j/tree/release/8.0) | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md)     |
```
Comment (qiancai) [inline develop/dev-guide-supported-by-pingcap.md:33]:
```suggestion
| Java       | [JDBC](https://dev.mysql.com/downloads/connector/j/)                     | 5.1.46；8.0.29  | Full | 5.1.46：N/A；8.0.29：[pingcap/mysql-connector-j](https://github.com/pingcap/mysql-connector-j/tree/release/8.0)                                                                                     | [TiDB 和 Java 的简单 CRUD 应用程序](/develop/dev-guide-sample-application-java.md)     |
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 14]
```diff
@@ -12,60 +12,65 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
+| Python                | [SQLAlchemy](https://www.sqlalchemy.org/)                                                                                                                                     | v1.4.37     | Beta | N/A                                                    | N/A                                                                            |
 
 ## 应用框架
 
-| 应用框架 |    数据接入框架     | 最新已测试版本 | 支持等级 | 教程地址 |
-| :------: | :-----------------: | :------------: | :------: | :------: |
-|  Spring  |        JDBC         |
-|    ⬆️    | JPA(with Hibernate) |
-|    ⬆️    |       MyBatis       |
+| 应用框架   | 数据接入框架              | 最新已测试版本 | 支持等级 | 教程地址 |
```
Comment (Oreoxmt) [inline develop/dev-guide-supported-by-pingcap.md:60]:
```suggestion
| 应用框架   | 数据接入框架              | 最新已测试版本 | 支持等级 | 教程 |
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 15]
```diff
@@ -12,60 +12,65 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
+| GUI                                           | 最新测试版本  | 支持级别 | 教程  |
+|-----------------------------------------------|---------|------|-----|
+| [DBeaver](https://dbeaver.io/)                | 22.1.0  | Beta | N/A |
+| [Navicat for MySQL](https://www.navicat.com/) | 16.0.14 | Beta | N/A |
 
-|                       IDE                        | 最新已测试版本 | 支持等级 | 教程地址 |
-| :----------------------------------------------: | :------------: | :------: | :------: |
-| [DataGrip](https://www.jetbrains.com/datagrip/)  |
-| [IntelliJ IDEA](https://www.jetbrains.com/idea/) |
+| IDE                                              | 最新已测试版本 | 支持等级 | 教程地址 |
```
Comment (Oreoxmt) [inline develop/dev-guide-supported-by-pingcap.md:73]:
```suggestion
| IDE                                              | 最新已测试版本 | 支持等级 | 教程 |
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 16]
```diff
@@ -12,60 +12,65 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
+| Spring | MyBatis             | N/A     | Beta | N/a  |
 
 ## GUI
 
-|                  GUI                   | 最新已测试版本 | 支持等级 | 教程地址 |
-| :------------------------------------: | :------------: | :------: | :------: |
-| [Navicat](https://www.navicat.com/en/) |
-
-## IDE
+| GUI                                           | 最新测试版本  | 支持级别 | 教程  |
```
Comment (Oreoxmt) [inline develop/dev-guide-supported-by-pingcap.md:68]:
```suggestion
| GUI                                           | 最新已测试版本  | 支持等级 | 教程  |
```

**Review by Oreoxmt — COMMENTED** [review]
@crazycs520 Please add this file to [TOC.md](https://github.com/pingcap/docs-cn/blob/master/TOC.md)

File: develop/dev-guide-supported-by-pingcap.md [hunk 17]
```diff
@@ -12,60 +12,65 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
 
 PingCAP 与开源社区合作，通过三方工具提供以下支持：
 
-- Full: 表明 PingCAP 将尽力支持该工具的绝大多数功能兼容性。将定期地对下表中记录的最新版本进行测试。
-- Beta: 表名 PingCAP 正在努力支持该工具。该工具的主要功能将与 TiDB 兼容(连接数据库及基本数据库操作)，但有可能会出现意外行为，且可能需要额外的步骤进行集成。
+- Full：表明 PingCAP 将尽力支持该工具的绝大多数功能兼容性。将定期地对下表中记录的最新版本进行测试。
```
Comment (Oreoxmt) [inline develop/dev-guide-supported-by-pingcap.md:15]:
```suggestion
- Full：表明 PingCAP 将尽力支持该工具的绝大多数功能兼容性，将定期地对下表中记录的最新版本进行测试。
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 18]
```diff
@@ -12,60 +12,65 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
...
+- Beta：表明 PingCAP 正在努力为该工具提供全面支持。该工具的主要功能可以和 TiDB 兼容（例如连接和数据库的基本操作），但缺乏对该工具所有功能的支持，有可能出现一些意外的行为。
+
+> **注意：**
+>
+> 除非明确说明，否则对于支持的 Driver 或者 ORM 框架并不包括[应用端事务重试和错误处理](/develop/dev-guide-transaction-troubleshoot.md#应用端重试和错误处理)。
 
 如果你在使用此处列出的工具连接 TiDB 时出现问题，请提出一个包含详细信息的[问题](https://github.com/pingcap/tidb/issues/new?assignees=&labels=type%2Fquestion&template=general-question.md)，以帮助在此工具的支持上得到进展。
```
Comment (Oreoxmt) [inline develop/dev-guide-supported-by-pingcap.md:22]:
```suggestion
如果在使用本文列出的工具连接 TiDB 时出现问题，请在 GitHub 上提交包含详细信息的 [issue](https://github.com/pingcap/tidb/issues/new?assignees=&labels=type%2Fquestion&template=general-question.md)，以帮助在此工具的支持上得到进展。
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 19]
```diff
@@ -12,60 +12,65 @@ TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 
 
 PingCAP 与开源社区合作，通过三方工具提供以下支持：
```
Comment (Oreoxmt) [inline develop/dev-guide-supported-by-pingcap.md:13]: - 删除 Line4 空行
- Line9 “将适配一组满足大多数语言和框架的工具，此页面上将展示这些工具和它们的支持等级。” 缺少主语

Comment (crazycs520) [reply develop/dev-guide-supported-by-pingcap.md:12]: done.

File: develop/dev-guide-supported-by-pingcap.md [hunk 20]
```diff
@@ -1,71 +1,75 @@
 ---
 title: PingCAP 维护的三方库
 aliases: ['/zh/tidb/dev/supported-by-pingcap']
-
 ---
 
 # PingCAP 维护的三方库
 
-TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 及其他工具与 TiDB 兼容。将适配一组满足大多数语言和框架的工具，此页面上将展示这些工具和它们的支持等级。
+TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 及其他工具与 TiDB 兼容，此页面上将展示这些工具和它们的支持等级。
```
Comment (Oreoxmt) [inline develop/dev-guide-supported-by-pingcap.md:8]:
```suggestion
TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 及其他工具与 TiDB 兼容。本文主要介绍这些工具和它们的支持等级。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Comment by Oreoxmt** [comment]
@winkyao PTAL

File: develop/dev-guide-supported-by-pingcap.md [hunk 21]
```diff
@@ -1,71 +1,75 @@
...
-TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 及其他工具与 TiDB 兼容。将适配一组满足大多数语言和框架的工具，此页面上将展示这些工具和它们的支持等级。
+TiDB 对 MySQL 协议的支持，使得大部分适配 MySQL 的 Driver、ORM 及其他工具与 TiDB 兼容。本文主要介绍这些工具和它们的支持等级。
 
 ## 支持等级
 
 PingCAP 与开源社区合作，通过三方工具提供以下支持：
 
-- Full: 表明 PingCAP 将尽力支持该工具的绝大多数功能兼容性。将定期地对下表中记录的最新版本进行测试。
-- Beta: 表名 PingCAP 正在努力支持该工具。该工具的主要功能将与 TiDB 兼容(连接数据库及基本数据库操作)，但有可能会出现意外行为，且可能需要额外的步骤进行集成。
+- Full：表明 PingCAP 将尽力支持该工具的绝大多数功能兼容性，将定期地对下表中记录的最新版本进行测试。
```
Comment (winkyao) [inline develop/dev-guide-supported-by-pingcap.md:14]:
```suggestion
- Full：表明 PingCAP 将尽力支持该工具的绝大多数功能兼容性，将定期地对下表中记录的最新版本进行测试。
```
```suggestion
- Full：表明 PingCAP 已经支持该工具的绝大多数功能兼容性，并且在新版本中对其保持兼容，将定期地对下表中记录的新版本进行兼容性测试。
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 22]
```diff
@@ -1,71 +1,75 @@
...
+- Beta：表明 PingCAP 正在努力为该工具提供全面支持。该工具的主要功能可以和 TiDB 兼容（例如连接和数据库的基本操作），但缺乏对该工具所有功能的支持，有可能出现一些意外的行为。
```
Comment (winkyao) [inline develop/dev-guide-supported-by-pingcap.md:15]:
```suggestion
- Beta：表明 PingCAP 正在努力为该工具提供全面支持。该工具的主要功能可以和 TiDB 兼容（例如连接和数据库的基本操作），但缺乏对该工具所有功能的支持，有可能出现一些意外的行为。
```
```suggestion
- Beta：表明 PingCAP 正在努力为该工具提供支持。该工具的主要功能可以被 TiDB 兼容，但缺乏对该工具作完整的兼容性验证，有可能出现一些意外的行为。
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 23]
```diff
@@ -1,71 +1,75 @@
...
+| Python                | [peewee](https://github.com/coleifer/peewee/)                                                                                                                                 | v3.14.10    | Beta | N/A                                                    | N/A                                                                            |
+| Python                | [PonyORM](https://ponyorm.org/)                                                                                                                                               | v0.7.16     | Beta | N/A                                                    | N/A                                                                            |
+| Python                | [SQLAlchemy](https://www.sqlalchemy.org/)                                                                                                                                     | v1.4.37     | Beta | N/A                                                    | N/A                                                                            |
 
 ## 应用框架
```
Comment (Icemap) [inline develop/dev-guide-supported-by-pingcap.md:58]:
```suggestion
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 24]
```diff
@@ -1,71 +1,75 @@
...
-| 应用框架 |    数据接入框架     | 最新已测试版本 | 支持等级 | 教程地址 |
-| :------: | :-----------------: | :------------: | :------: | :------: |
-|  Spring  |        JDBC         |
-|    ⬆️    | JPA(with Hibernate) |
-|    ⬆️    |       MyBatis       |
+| 应用框架   | 数据接入框架              | 最新已测试版本 | 支持等级 | 教程 |
+| ------ | ------------------- | ------- | ---- | ---- |
+| Spring | JDBC                | N/A     | Beta | N/A  |
+| Spring | JPA(with Hibernate) | N/A     | Beta | N/A  |
+| Spring | MyBatis             | N/A     | Beta | N/a  |
```
Comment (Icemap) [inline develop/dev-guide-supported-by-pingcap.md:64]:
```suggestion
```

File: develop/dev-guide-supported-by-pingcap.md [hunk 25]
```diff
@@ -1,71 +1,66 @@
...
+| Python                | [peewee](https://github.com/coleifer/peewee/)                                                                                                                                 | v3.14.10    | Beta | N/A                                                    | N/A                                                                            |
+| Python                | [PonyORM](https://ponyorm.org/)                                                                                                                                               | v0.7.16     | Beta | N/A                                                    | N/A                                                                            |
+| Python                | [SQLAlchemy](https://www.sqlalchemy.org/)                                                                                                                                     | v1.4.37     | Beta | N/A                                                    | N/A                                                                            |
 ## GUI
```
Comment (Icemap) [inline develop/dev-guide-supported-by-pingcap.md:56]:
```suggestion

## GUI
```

**Review by Icemap — APPROVED** [review]
LGTM

**Review by winkyao — APPROVED** [review]
LGTM

**Pull request merged**